### PR TITLE
Docstring additions and typo fixes

### PR DESCRIPTION
**Pull request by rsiemens**
improve code readability in _parse_decls

### CODE REVIEW

**Comment by untitaker** [comment]
Some typo fixes are legit, but some things are not necessary. `and-or` is just a style choice.

File: click/core.py [hunk 1]
```diff
@@ -1425,7 +1425,7 @@ class Option(Parameter):
     :param show_default: controls if the default value should be shown on the
                          help page.  Normally, defaults are not shown.
     :param prompt: if set to `True` or a non empty string then the user will
-                   be prompted for input if not set.  If set to `True` the
+                   be prompted for input.  If set to `True` the
```
Comment (untitaker) [inline click/core.py]: This sentence still doesn't make sense.

**Comment by untitaker** [comment]
@Nudies Any update?

**Comment by Nudies** [comment]
@untitaker Apologies seems I missed your original comment 27days ago. I reverted back the `and-or` change, sorry for making that assumption. Also cleaned up the `Option` class docstring a little.

File: click/core.py [hunk 2]
```diff
@@ -1424,9 +1424,9 @@ class Option(Parameter):
 
     :param show_default: controls if the default value should be shown on the
                          help page.  Normally, defaults are not shown.
-    :param prompt: if set to `True` or a non empty string then the user will
-                   be prompted for input if not set.  If set to `True` the
-                   prompt will be the option name capitalized.
+    :param prompt: if set to `True` or a non empty string then the user will be
+                   prompted for input.  If set to `True` the prompt will be the
+                   option name capitalized.
```
Comment (Nudies) [inline click/core.py]: I cleaned this up a little. Is the sentence still confusing? If so what part?

**Comment by untitaker** [comment]
Cool, thanks!

**Pull request merged**